_data-prepper/trace-analytics.md

```diff
@@ -0,0 +1,312 @@
+
+* *otel-v1-apm-span* -  This index stores the output from [otel-trace-raw-processor](../data-prepper-plugins/otel-trace-raw-processor/README.md).
+* *otel-v1-apm-service-map* - This index stores the output from the [service-map-processor](../data-prepper-plugins/service-map-stateful/README.md).
+
+## Pipeline configuration
+
+### Trace analytics pipeline example
+
+The following example demonstrates how to build a pipeline that supports the [Trace Analytics OpenSearch Dashboards plugin]({{site.url}}{{site.baseurl}}/observability-plugin/trace/ta-dashboards/). This pipeline takes data from the OpenTelemetry Collector and uses two other pipelines as sinks. These two separate pipelines index trace and the service map documents for the dashboard plugin.
+
+Starting from Data Prepper 2.0, Data Prepper no longer supports `otel_trace_raw_prepper` processor due to the Data Prepper internal data model evolution. 
+Instead, users should use `otel_trace_raw`.
+
+```yml
+entry-pipeline:
+  delay: "100"
+  source:
+    otel_trace_source:
+      ssl: false
+  buffer:
+    bounded_blocking:
+      buffer_size: 10240
+      batch_size: 160
+  sink:
+    - pipeline:
+        name: "raw-pipeline"
+    - pipeline:
+        name: "service-map-pipeline"
+raw-pipeline:
+  source:
+    pipeline:
+      name: "entry-pipeline"
+  buffer:
+    bounded_blocking:
+      buffer_size: 10240
+      batch_size: 160
+  processor:
+    - otel_trace_raw:
+  sink:
+    - opensearch:
+        hosts: ["https://localhost:9200"]
+        insecure: true
+        username: admin
+        password: admin
+        index_type: trace-analytics-raw
+service-map-pipeline:
+  delay: "100"
+  source:
+    pipeline:
+      name: "entry-pipeline"
+  buffer:
+    bounded_blocking:
+      buffer_size: 10240
+      batch_size: 160
+  processor:
+    - service_map_stateful:
+  sink:
+    - opensearch:
+        hosts: ["https://localhost:9200"]
+        insecure: true
+        username: admin
+        password: admin
+        index_type: trace-analytics-service-map
+```
+
+To maintain similar ingestion throughput and latency, scale the `buffer_size` and `batch_size` by the estimated maximum batch size in the client request payload.
+{: .tip}
+
+
+#### Example: otel trace
+Example `otel-trace-source` with SSL and Basic Authentication enabled. Note that you will have to change your `otel-collector-config.yaml` accordingly:
+
+```yaml
+source:
+  otel_trace_source:
+    #record_type: event  # Add this when using Data Prepper 1.x. This option is removed in 2.0
+    ssl: true
+    sslKeyCertChainFile: "/full/path/to/certfile.crt"
+    sslKeyFile: "/full/path/to/keyfile.key"
+    authentication:
+      http_basic:
+        username: "my-user"
+        password: "my_s3cr3t"
+```
+
+
+#### Example: pipeline.yaml
+Example `pipeline.yaml` without SSL and Basic Authentication for the `otel-trace-source`:
+
+```yaml
+otel-trace-pipeline:
```

Comment:
We should probably rename this to `entry-pipeline` for consistency.

Reply:
@dlvenable To confirm, should `otel-trace-pipeline` be changed to `entry-pipeline`? Or are you referring to another item?